chore(deps): replace chalk with picocolors

Fuzzyma opened the pull request:
Replaced `chalk` with `picocolors` as discussed in #1340.
I ended up using picocolors to support the NO_COLOR env (https://no-color.org/)

**Checklist**:

<!-- add "N/A" to the end of each line that's irrelevant to your changes -->

<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->

- [ ] Documentation added to the [docs site](https://github.com/testing-library/testing-library-docs) N/A
- [ ] Tests
- [ ] TypeScript definitions updated N/A
- [ ] Ready to be merged

<!-- feel free to add additional comments -->

I wasnt able to run the tests. It errored with `TypeError: _browserslist.findConfigFile is not a function`.
I dont want to call this mergable unless I can see the green tests.

// EDIT: The CI seems to fail with the same error

// OFFTOPIC: I saw that `@testing-library/jest-dom` is also using chalk but only in a test. However, it is listed as dependency. Maybe we can move it to devDependencies and also replace it?

eps1lon commented:
>  I saw that @testing-library/jest-dom is also using chalk but only in a test. However, it is listed as dependency. Maybe we can move it to devDependencies and also replace it?

It's listed in `devDependencies` as far as I can tell.

Fuzzyma commented:
@eps1lon I looked into it and this is a deadlock. I tried several combinations of pinned dependencies and got the tests to work eventually. But now `validate` fails.

I really wonder why nobody ever bothered to commit a lock-file. Thats exactly what it is for :D.

The reason this is all not working is, is because some babel dependencies that are brought in require at least babel 7.22. However, that version calls browserlist with a different api (and therefore fails). Installing older versions of those packages gets rid of that error but now the validate fails with (to me unknown) reason because it complains that a function isnt found (which is clearly there).

So I have to give up for now. What other possibilities to we have?
Does anyway of you have a working setup? If so, please dump all the exact versions of your dependencies by (deleting any shrinkwrap and lock file and then) running

```
npm shrinkwrap
```

Give me that file :D

Fuzzyma commented:
OMG I DID IT!!!!
I had to commit the package-lock.json to convince the CI to install the correct versions.
So not sure what to do with this.

It would be really beneficial to get rid of old supported browsers so we can update browserslist so we can update babel so we can get rid of all that overwrites and really unstable depenceny situation

timdeschryver commented on .gitignore:
```diff
@@ -5,5 +5,4 @@ dist
 
 # these cause more harm than good
 # when working with contributors
-package-lock.json
```
> I had to commit the package-lock.json to convince the CI to install the correct versions. So not sure what to do with this.

I might be wrong about this, but we're not a big fan of introducing lock files (https://github.com/testing-library/dom-testing-library/pull/624).
That's why we're also not using it during our CI pipeline:
https://github.com/testing-library/dom-testing-library/blob/a86c54ccda5242ad8dfc1c70d31980bdbf96af7f/.github/workflows/validate.yml#L49

If you want you can revert this change, and I can give it a look in order to fix this.

MichaelDeBoey replied:
This originates from @kentcdodds' package starters.

If I remember correctly (but Kent can probably go more in depth here) the main reason is for when different people are adding/removing/updating different (dev)dependencies at the same time.
When merging one PR, the other one could automatically be in conflict with the one just merged, which is a pain to resolve.

MichaelDeBoey replied:
A good solution (for now) would probably be to pin the versions to an exact number and investigate this further in another PR, so we don't hold back this PR

Fuzzyma replied:
That is what I tried exactly. I pinned the versions but the pinned dependencies also have dependencies and they are not pinned. The version I have locally runs for whatever reason. So my set of dependencies and sub dependencies work together.
I spend quite a while to find a set of overwrites and versions that ran at all. So it might take another while to find out what I have installed vs what the ci is installing :D.

Yes, package-lock conflicts are annoying. But npm can handle merge conflicts nowadays and will resolve them for you if you just run "npm install".

We can also move this discussion into its own issue as you suggested. I think it would be worthwhile to decide that first before putting more time into investigating the exact dependencies we need to overwrite :D

timdeschryver commented on package.json:
```diff
@@ -59,29 +59,33 @@
+    "lz-string": "1.5.0",
+    "picocolors": "1.1.1",
+    "pretty-format": "27.0.2",
+    "typescript": "4.5"
```
`typescript` needs to be listed as a `devDependency`, or is there a reason to move it into the dependencies?

Fuzzyma replied:
Just an oversight and probably a leftover from trying to get the versions right :D

timdeschryver reviewed (changes requested):
Thanks for the pull request @Fuzzyma !
I did leave some comments, feel free to ask for assistance if needed.

Fuzzyma commented:
@timdeschryver I reverted the commit of the package-lock and the ci is red again. So would love some help here. Clearly the CI does _something_ with the lock-file :D

Also moved typescript back to devDependencies where it rightfully belongs

Fuzzyma commented:
It seems like that there was a to introduce a lockfile in 2022 already but Kent wanted the aproval from one other maintainer (https://github.com/testing-library/dom-testing-library/pull/624#issuecomment-643260300). Maybe that could be you? :D



Kent also seems to use lockfiles now:
https://github.com/kentcdodds/kentcdodds.com

MichaelDeBoey commented:
@Fuzzyma Kent's not opposed to using lock files in projects/apps, but the way I understood it he is opposed using them in packages

timdeschryver commented:
I will take a look at this in the next days (I might push directly into this branch). 
AFAIK @MichaelDeBoey is right about the reasons of why not to include a lock file, additionally it also allows people to use their package manager of choice. We might revisit this later, but I prefer not to do this within this PR.

timdeschryver reviewed (approved):
Pinning `@babel/helper-compilation-targets` seems to do the trick.
I used the last version of a successful build.

timdeschryver commented:
@eps1lon does this seem good to you? 
As mentioned in the comments, `@babel/helper-compilation-targets` was pinned to make the build happy without upgrading deps/changing the browserlist support (as was done in #1346)

benmccann commented:
This has three approvals. Could we merge it?

MatanBobi commented:
Thanks @benmccann for the ping.
I've just merged https://github.com/testing-library/jest-dom/pull/659, let's see we're not seeing any regressions there and I'll merge this one un the upcoming week :)
